website/server/controllers/api-v3/members.js

```diff
@@ -402,6 +362,109 @@ function _getMembersForItem (type) {
   };
 }
 
+async function _getMemberTasksFromChallenge (challengeId, memberId, user, res) {
+  const member = await User.findById(memberId).select(`${nameFields} challenges`).exec();
+  if (!member) throw new NotFound(res.t('userWithIDNotFound', { userId: memberId }));
+
+  const challenge = await Challenge.findById(challengeId).exec();
+  if (!challenge) throw new NotFound(res.t('challengeNotFound'));
+
+  // optionalMembership is set to true because even if you're
+  // not member of the group you may be able to access the challenge
+  // for example if you've been booted from it, are the leader or a site admin
+  const group = await Group.getGroup({
+    user, groupId: challenge.group, fields: '_id type privacy', optionalMembership: true,
+  });
+  if (!group || !challenge.canView(user, group)) throw new NotFound(res.t('challengeNotFound'));
+  if (!challenge.isMember(member)) throw new NotFound(res.t('challengeMemberNotFound'));
+
+  const challengeTasks = await Tasks.Task.find({
+    userId: memberId,
+    'challenge.id': challengeId,
+  })
+    .select('-tags') // We don't want to return the tags publicly TODO same for other data?
+    .exec();
+
+  return challengeTasks.map(task => {
+    task.checklist = []; // Clear checklists as they are private
```

Comment:
I know this is old code but I was wondering if maybe instead of manually removing the checklist data it wouldn't be easier to simply not return it from the database like it's done for tags